### DISCUSSION TITLE
Result of calculation is not updated

### DISCUSSION THREAD

**Discussion by donacrisante** [neuefische/questions#684] [Web] [answered]
Hello everyone, I am again facing an error that I can't fix. When I enter a new journey and click on "calculate your impact", the CO2 emission now remains the same for each entry and is not updated depending on which means of transport was selected. Also, the result stays at 92.75kg even if I reload the page. Can anyone help me with this? Many thanks in advance! 

<img width="509" alt="Bildschirm­foto 2023-08-21 um 10 58 44" src="https://github.com/neuefische/questions/assets/128093782/0e4a6867-3127-425a-b4e1-5d24851dcbf7">

Here my PR:
https://github.com/donacrisante/Capstone-project/pull/10

**Comment by Kheirah** [comment, marked as answer]
I looked into it. As the app's state is architected right now, you will always get the same CO2 result for every journey and it will always be the last CO2 that was calculated, because: the state "result" is a simple number and it only holds one value at a time - the last one.

To fix this, you need to remove the `result` state entirely and add a new attribute to your `entries` array of objects. Call it "results" or "co2" or whatever you like. In that new attribute you can store the calculated co2. And then every entry will have their own co2 value which you can display.

For that you will need to refactor your code quite a bit. You don't need to pass the `result` and `setResult` down the hierarchy of components anymore. Instead you will have to pass `entries` and `setEntries` and use the new attribute within your entries' objects to display the co2 value and to save to it when you calculate a new value. Better yet, you could write a handle function in `_app.js` which takes a parameter (the calculated value) and adds it to your state. That function you then pass down (instead of the `setEntries` function).
